### ISSUE TITLE
Crash generating sys for xapp

### ISSUE THREAD

**Issue by ItzSwirlz** [gtk-rs/gir#1203]
mouse broke, so for now have a picture, since terminal is annoying copying. I pushed the source to https://github.com/Ubuntu-Cinnamon-Remix/xapp-rust. Fedora 34
![Screenshot from 2021-07-23 19-06-18](https://user-images.githubusercontent.com/24401303/126849205-75fd5bdb-f653-40f1-abc5-90bcb54c4c97.png)

**Comment by bilelmoussaoui** [comment]
The issue is caused by your GIR file, it needs a `name` in the `package` tag. It would look like this `<package name="gobject-2.0"/>` at the top of the file.

**Comment by sdroege** [comment]
We should maybe provide a better error message here instead of just panicking

**Comment by ItzSwirlz** [comment]
Yeah. I had no idea about the issue of HarfBuzz until GirXml was run. It should be run again on the gir we are generating
